File: frontend/types/planning.ts
export interface PlanImportResult {
  plan_type: string;
  version_name: string;
  total_rows: number;
  inserted_rows: number;
  invalid_rows: number;
  issues: Array<{ row_number?: number; message: string }>;
}
export interface VarianceLine {
  period: string;
  reporting_group: string;
  actual: string | number;
  budget: string | number;
  forecast: string | number;
  budget_variance: string | number;
  budget_variance_percent?: string | number | null;
  forecast_variance: string | number;
  forecast_variance_percent?: string | number | null;
}

File: backend/app/schemas/finance/planning.py
from datetime import date
from decimal import Decimal
from pydantic import BaseModel, Field


class PlanImportResponse(BaseModel):
    plan_type: str
    version_name: str
    total_rows: int
    inserted_rows: int
    invalid_rows: int
    issues: list[dict] = Field(default_factory=list)


class PlanLineResponse(BaseModel):
    period: date
    plan_type: str
    version_name: str
    reporting_group: str
    reporting_subgroup: str | None = None
    source_account_code: str | None = None
    branch_id: str | None = None
    amount: Decimal


class VarianceLineResponse(BaseModel):
    period: date
    reporting_group: str
    actual: Decimal
    budget: Decimal
    forecast: Decimal
    budget_variance: Decimal
    budget_variance_percent: Decimal | None
    forecast_variance: Decimal
    forecast_variance_percent: Decimal | None
